src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -105,12 +105,82 @@ uct_cuda_ipc_md_query(uct_md_h md, uct_md_attr_v2_t *md_attr)
     return UCS_OK;
 }
 
+static ucs_status_t
+uct_cuda_ipc_mem_reg_push_ctx(CUdeviceptr address, CUdevice *cuda_device_p,
+                              int *is_ctx_pushed)
+{
+#define UCT_CUDA_IPC_NUM_ATTRS 2
+    CUdevice cuda_device = CU_DEVICE_INVALID;
+    CUpointer_attribute attr_type[UCT_CUDA_IPC_NUM_ATTRS];
+    void *attr_data[UCT_CUDA_IPC_NUM_ATTRS];
+    CUcontext cuda_ctx, cuda_curr_ctx;
+    int cuda_device_ordinal;
+    ucs_status_t status;
+
+    *is_ctx_pushed = 0;
+
+    attr_type[0] = CU_POINTER_ATTRIBUTE_CONTEXT;
+    attr_data[0] = &cuda_ctx;
+    attr_type[1] = CU_POINTER_ATTRIBUTE_DEVICE_ORDINAL;
+    attr_data[1] = &cuda_device_ordinal;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(
+            cuPointerGetAttributes(UCT_CUDA_IPC_NUM_ATTRS, attr_type, attr_data,
+                                   address));
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    if (cuda_ctx == NULL) {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGet(&cuda_device,
+                                                      cuda_device_ordinal));
+        if (status != UCS_OK) {
+            goto out;
+        }
+
+        status = uct_cuda_primary_ctx_retain(cuda_device, 0, &cuda_ctx);
+        if (status != UCS_OK) {
+            goto out;
+        }
+    } else {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&cuda_curr_ctx));
```

Comment:
This will fail if context is not set and return error. This PR should not fail but set the desired context.

Reply:
Please ignore the comment. It will return UCS_OK.